doc/changelog.rst

```diff
@@ -1,10 +1,102 @@
 Changelog
 =========
 
+0.3.1
+-----
+
+Released on June 23, 2022
+
+Description
+
+ - Updated and improved the documentation of the SmartRedis library. In particular, a new SmartRedis Integration Guide provides an introduction to using the SmartRedis library and integrating it with existing software. (PR261_) (PR260_) (PR259_) (SSPR214_)
+
+ - Implemented aggregation of lists of DataSets for all client languages, enabling the SmartRedis library to perform pipelining of operations. (PR258_) (PR257_) (PR256_) (PR248_) New commands are:
+  - append_to_list()
+  - delete_list()
+  - copy_list()
+  - rename_list()
+  - get_list_length()
+  - poll_list_length()
+  - poll_list_length_gte()
+  - poll_list_length_lte()
+  - get_datasets_from_list()
+  - get_dataset_list_range()
+  - use_list_ensemble_prefix()
+
+ - Implement multithreaded execution of parallel tasks, including all multi-shard operations. The number of threads devoted for this purpose is controlled by the new environment variable SR_THERAD_COUNT. The value defaults to 4, but may be any positive integer or special value zero, which will cause the SmartRedis runtime to allocate one thread for each available hardware context. (PR251_) (PR246_)
+
+ - Augment support for GPUs by implementing multi-GPU convenience functions for all client languages. (PR254_) (PR250_) (PR244_) New commands are:
+  - set_model_from_file_multigpu()
+  - set_model_multigpu()
+  - set_script_from_file_multigpu()
+  - set_script_multigpu()
+  - run_model_multigpu()
+  - run_script_multigpu()
+  - delete_model_multigpu()
+  - delete_script_multigpu()
+
+ - Added API calls for all clients to delete models and scripts from the backend database. (PR240_) New commands are:
+  - delete_script()
+  - delete_model()
+
+ - Added clustered Redis testing to automated GitHub check-in testing. (PR239_)
```

Comment:
Maybe the right way to organize this is by theme by either the portion of the code (e.g. 'Fortran Client') or by type of change (e.g. bug fix, performance enhancements, quality of life).